### ISSUE TITLE
minimize total loss without step loss

### ISSUE THREAD

**Issue by leiup** [tensorflow/tensorflow#47461]
I want to minimize total loss without step loss (each loss no back-propagation), how to implement it?    
total_loss = 0
idx = 0
for data in train_datas:
       pre = model(data)
       loss = metric(pre, gt)
       total_loss = total_loss + loss
       idx = idx + 1
       if idx %10 == 0:
              train_ops = optimizer(total_loss)
              sess.run(train_ops)

**Comment by Saduf2019** [comment]
I am able to replicate the issue reported on tf 2.4 and nightly, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/0fa70861612a9e33b4ba427d1e506971/untitled554.ipynb).

**Comment by leiup** [comment]
@Saduf2019 

I want to minimize total loss without step loss (each loss no back-propagation), how to implement it?
total_loss = 0
idx = 0
for data in train_datas:
  pre = model(data)
  loss = metric(pre, gt)
  total_loss = total_loss + loss
  idx = idx + 1
  if idx %10 == 0:
    train_ops = optimizer(total_loss)
    sess.run(train_ops)
